pandas/io/parsers/base_parser.py

```diff
@@ -1148,6 +1148,10 @@ def _isindex(colspec):
             )
 
             new_data[new_name] = col
+
+            if len(colspec) == 1:
```

Comment:
this seems like very special casey here

Reply:
It is, yes. We can only keep the original column formatted as date, if it represents the whole date. In cases where two columns combined represent a date we can only keep the original format

Reply:
ok worth a comment about this then for future readers

Reply:
Done